File: C/Algorithms/766.c
#include <stdbool.h>

bool isToeplitzMatrix(int **mat, int matrixSize, int *matrixColSize) {
  int cols = *matrixColSize - 1;
  for (int i = 0; i < matrixSize - 1; i++) {
    for (int j = 0; j < cols; j++) {
      if (mat[i][j] != mat[i + 1][j + 1]) {
        return false;
      }
    }
  }
  return true;
}

File: Python/Algorithms/766.py
class Solution:
    def isToeplitzMatrix(self, mat):
        rng_i, rng_j = range(len(mat)-1), range(len(mat[0])-1)
        for i in rng_i:
            for j in rng_j:
                if mat[i][j] != mat[i+1][j+1]:
                    return False
        return True
